proposals/20210525-feast_example_gen.md

```diff
@@ -39,15 +39,38 @@ There is no potential impact, overlap, or synergies with other projects. Depende
 
 ## Project Implementation
 
-* At a high level, there are 3 major APIs [3]:
-1. Connect to a remote Feast instance. The connection string could be part of a general procedure to configure the feature store using a JSON/YAML file.
-2. Provide a wrapper for `get_online_features`. Feast returns an `OnlineResponse` [5], which will be converted to a tensor.
-3. Provide a wrapper for `get_historical_features`. Feast returns a `RetrievalJob` [6], which will be used to materialize the results and convert them to a tensor.
+The Feast connector will be implemented as a custom `ExampleGen` component. It will inherit from `QueryBasedExampleGen`. It will ultimately closely resemble `PrestoExampleGen` [8]. `FeastExampleGen` will emit `tf.Example` records.
 
-* Important Considerations
+```python
+from tfx.examples.custom_components.feast_example_gen.proto import feast_config_pb2
+from tfx.examples.custom_components.feast_example_gen.feast_component.component import FeastExampleGen
+
+feast_config = feast_config_pb2.FeastConnConfig(host=.., port=..)
```

Comment:
The current Feast implementation captures all configuration within a `feature_store.yaml` file, and doesn't require an API to connect to a feature store. The feature store registry is completely encapsulated within a protobuf binary that we store either locally or on GCS/S3.

Is this `feast_config` pointing to something else?

Reply:
FeastConnConfig (modeled after [`PrestoConnConfig`](https://github.com/tensorflow/tfx/blob/983794c2ec2ca567dddffab0e4827fa29bcb2230/tfx/examples/custom_components/presto_example_gen/proto/presto_config.proto)) is a protobuf spec for the connection configs only. We would not want users to interact with [feature_store.yaml](https://docs.feast.dev/reference/feature-repository/feature-store-yaml) at all as that controls other (potentially sensitive) [RepoConfig](https://rtd.feast.dev/en/latest/#feast.repo_config.RepoConfig) knobs.

`FeastConnConfig` would only need a few standard fields: `host`, `port`, `catalog`? (although I am not sure how the concept of catalogs would fit with Feast's concept of [feature views](https://docs.feast.dev/concepts/feature-view). Probably just best to start with `host` and `port` and the basic HTTP client options like timeout etc).

https://docs.feast.dev/feast-on-kubernetes/getting-started/connect-to-feast/python-sdk
```python
# Connect to an existing Feast Core deployment
client = Client(core_url='feast.example.com:6565')
```